File: src/problemset/array/PascalsTriangle.py
''' -*- coding: utf-8 -*-
date: 2020-04-01
time: 15:00:55
author: MoeexT
problem: 118.
url: List<Integer> row = new ArrayList<>();
'''
from typing import List


class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        '''
        time: 28ms 92.81%
        mem: 13.7MB 5.12%
        '''
        if numRows == 0:
            return []
        res = [[1]]
        for i in range(1, numRows):
            row = [1]
            for j in range(1, i):
                row.append(res[-1][j-1] + res[-1][j])
            row.append(1)
            res.append(row)
        return res


if __name__ == "__main__":
    solution = Solution()
    print(solution.generate(1))

File: src/problemset/array/PascalsTriangle.java
package problemset.array;

import java.util.ArrayList;
import java.util.List;

/** 
 * @date: 2020-04-01
 * @time: 14:33:19
 * @author: MoeexT
 * @problem: 118
 * @url: https://leetcode-cn.com/problems/pascals-triangle/
 */

public class PascalsTriangle {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> list = new ArrayList<>();
        if (numRows == 0) {
            return list;
        } else if (numRows == 1) {
            list.add(new ArrayList<Integer>(1));
            return list;
        }

        for (int i = 1; i <= numRows; i++) {
            List<Integer> row = new ArrayList<>();
            row.add(1);
            for (int j = 1; j < i; j++) {
                
            }
            row.add(1);
            list.add(row);
        }
        
        return list;
    }

    public static void main(String[] args) {
        PascalsTriangle obj = new PascalsTriangle();
        System.out.println(obj.generate(5));
    }
}
